gym/vector/async_vector_env.py

```diff
@@ -648,7 +648,7 @@ def _worker(index, env_fn, pipe, parent_pipe, shared_memory, error_queue):
 
             elif command == "step":
                 observation, reward, done, info = env.step(data)
-                if done:
+                if bool(done):
```

Comment:
I don't think this actually does anything, the conditional works even with an integer:
```python
if 2:
    print("Hi")
> Hi
```

An argument might be made for readability, but I don't think casting to bool actually makes it more readable

Reply:
The fact that `if done` still works is actually a pretty good showcase of the (mostly) backwards-compatibility of this

Reply:
> An argument might be made for readability, but I don't think casting to bool actually makes it more readable

Yes, I put this for readability. This explicit type casting is made for _"keep in mind that `expression` is not a boolean in statement `bool(expression)`"_.